### ISSUE TITLE
[Bug] Qwen3-ASR silently truncates audio longer than 30 seconds

### ISSUE THREAD

**Issue by lijrjyan** [sgl-project/sglang-omni#1173]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. I have included the environment information and a minimal reproducible example.
- [x] 4. This is a bug report rather than a usage question.
- [x] 5. This issue is written in English.

### Describe the bug

Qwen3-ASR on current `main` silently discards audio after the first 30 seconds. The request still returns HTTP 200 and reports the original full duration in `usage`, so callers cannot tell that the transcription is incomplete.

The request builder calls `WhisperFeatureExtractor` with `padding="longest"` and `truncation=True`. On the 62.455-second reproduction file, this produces 3,000 valid mel frames. Setting only `truncation=False` produces 6,245 valid mel frames.

The official `qwen-asr==0.0.6` Transformers backend uses the same model weights and the same `WhisperFeatureExtractor`, but explicitly sets `truncation=False`. It transcribed the complete 62.455-second file in three out of three runs. This indicates that 30 seconds is a limit of the current SGLang-Omni preprocessing path rather than a Qwen3-ASR model capability limit.

Potentially related paths, identified by source inspection but not yet validated end to end:

- Whisper ASR, including Whisper Large v3, calls its feature extractor without overriding the default truncation behavior. It may therefore exhibit the same silent truncation for uploads longer than 30 seconds. This has not yet been reproduced with a running Whisper server.
- TTS and Omni quality evaluations that use Qwen3-ASR as the WER transcriber may indirectly ignore speech after 30 seconds when a generated sample exceeds that duration. This may affect long-form or max-token outliers from Qwen3-TTS, Higgs Audio, MOSS-TTS, Voxtral TTS, FishAudio, and Qwen3-Omni talker evaluations. The potential downstream impact has not yet been quantified.
- Fun-ASR Nano has a related 30-second implementation boundary, but it explicitly rejects longer audio instead of silently returning a partial transcript. It is therefore not confirmed to have this bug.
- MOSS-Transcribe-Diarize uses chunk-aware processor outputs, Qwen3-Omni's Hugging Face processor defaults to `truncation=False`, and Ming-Omni computes mel features from the complete waveform. Source inspection does not currently indicate the same truncation behavior in those paths.

### Reproduction

Start Qwen3-ASR from source:

```bash
sgl-omni serve \
  --config examples/configs/qwen3_asr_rtx4090.yaml \
  --model-path /path/to/Qwen3-ASR-1.7B \
  --model-name Qwen/Qwen3-ASR-1.7B \
  --host 127.0.0.1 \
  --port 8000 \
  --log-level debug
```

Download the single validation sample from `distil-whisper/librispeech_long` and submit its unmodified 62.455-second WAV:

```bash
curl -F file=@librispeech_long.wav \
  -F model=Qwen/Qwen3-ASR-1.7B \
  -F language=en \
  http://127.0.0.1:8000/v1/audio/transcriptions
```

Input provenance:

```text
Dataset: distil-whisper/librispeech_long
Dataset file: clean/validation-00000-of-00001-913508124a40cb97.parquet
Audio duration: 62.455 seconds
Audio sample rate: 16,000 Hz
Audio channels: 1
Audio SHA256: 8f418821d760804084ce9e054474098768b90c3e873489891f61162ba25ddfe3
```

Observed on SGLang-Omni `main`, repeated three times:

```text
HTTP status: 200
usage.seconds: 63
input feature shape: [1, 128, 3000]
feature attention-mask sum: 3000
transcript ending: "...and can discover."
```

All three responses were identical and stopped at the same incomplete sentence near the 30-second boundary.

Feature-extractor comparison on the same waveform:

```text
padding="longest", truncation=True:
  shape=[1, 128, 3000], attention_mask.sum()=3000

padding="longest", truncation=False:
  shape=[1, 128, 6245], attention_mask.sum()=6245
```

The official `qwen-asr==0.0.6` Transformers backend returned the complete transcript in three out of three runs. Its transcript continued after the SGLang stopping point and ended with:

```text
...before he says, like a shampooer in a Turkish bath, "Next man."
```

### Environment

```text
Operating system: Ubuntu 24.04.3 LTS
Kernel: 6.8.0-65-generic
Python: 3.12.3
SGLang-Omni commit: cf61f234e5a2cc7f63a24bdef84b446bd9a42f74
Qwen3-ASR model revision: 7278e1e70fe206f11671096ffdd38061171dd6e5
PyTorch: 2.11.0+cu130
PyTorch CUDA build: 13.0
Transformers: 5.6.0
NVIDIA driver: 580.95.05
GPU: NVIDIA GeForce RTX 5090
GPU memory: 32,607 MiB
CPU: AMD Ryzen 9 9950X 16-Core Processor
```

The official-engine comparison used:

```text
qwen-asr: 0.0.6
Transformers: 4.57.6
PyTorch: 2.11.0+cu130
Model revision: 7278e1e70fe206f11671096ffdd38061171dd6e5
```

Related pull request: [sgl-project/sglang-omni#1154 — feat(qwen3-asr): add RTX 4090 serving profile](https://github.com/sgl-project/sglang-omni/pull/1154) converts the silent truncation into an HTTP 400 response at 30 seconds, but does not enable the model's existing variable-length path.

**Comment by HowardZorn** [comment]
30 seconds is the default truncation time for OpenAI’s `/v1/audio/transcriptions` when using a model with speaker diarization capabilities ([gpt-4o-transcribe-diarize](https://developers.openai.com/api/docs/guides/speech-to-text#identify-speakers)); in this case, Qwen3-ASR likely should not adopt this parameter.

If we need to consider implementing a chunking strategy, we may need to utilize some of the existing conventions of `/v1/audio/transcriptions`.
